### DISCUSSION TITLE
@onready that's called after base class @onready

### DISCUSSION THREAD

**Discussion by feois** [godotengine/godot-proposals#9025] [Scripting]
Could be a boolean toggle like `@onready(after := false)`

**Comment by Mickeon** [comment]
What do you need this for? Most of the times, it's not necessary to assign before `_ready()` is called, and when it is, it's usually better to be explicit about it and assign the variable within `_ready()` itself.

**Reply by feois** [reply]
```gdscript
# Base class

@onready var x := MyClassX.new()
@onready var y := MyClassY.new(x)
@onready var z := MyClassZ.new(x, y)

# Subclass

# not working
@onready var y2 := MyClassY.new(x)
@onready var z2 := MyClassZ.new(x, y)
```
of course i can just do it in `_ready()` but `@onready` is just way cleaner

**Comment by theraot** [comment]
This is making me think that the derived class onready should always go after the base class onready. Because the derived class onready can depend on the base onready. But at no point the base class onready can depend on the derived class onready.

Is there any case where having the onready of the derived class run after would cause problems?

**Reply by AThousandShips** [reply]
This is implemented [here](
https://github.com/godotengine/godot/blob/63d6bda8e95ac992da74e84b2f3be62f3d85190b/modules/gdscript/gdscript.cpp#L1892-L1897), looking through the history but I suspect it's for simplicity, to not have to implement some stack to call it in that order

Implemented in:
* https://github.com/godotengine/godot/pull/62255

To solve a different issue where onready variables weren't ready by `_ready`

**Comment by dalexeev** [comment]
The current behavior was considered a bug and fixed for future 4.3:

* godotengine/godot#63329
* godotengine/godot#91364
